### PR TITLE
HTS-1208: Log Get Account response from NS&I in Splunk

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosave/connectors/HelpToSaveProxyConnector.scala [hunk 1]
```diff
@@ -142,6 +144,10 @@ class HelpToSaveProxyConnectorImpl @Inject() (http:              HttpClient,
             val _ = timerContext.stop()
             response.status match {
               case Status.OK ⇒
+
+                val path = s"/help-to-save/$nino/account?nino=$nino&systemId=$systemId&correlationId=$correlationId"
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosave/connectors/HelpToSaveProxyConnector.scala]: note that this is the url of the incoming request but not the NSI get account url

Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosave/connectors/HelpToSaveProxyConnector.scala]: it'll be more futureproof to pass the path in and get it from the `request` in the controller - you can find the full URI including the query parameters under `request.uri`

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/connectors/HelpToSaveProxyConnector.scala]: fixed

File: app/uk/gov/hmrc/helptosave/models/HTSEvent.scala [hunk 2]
```diff
@@ -154,3 +154,16 @@ object AccountCreated {
...
+
+case class GetAccountResult(nino: String, account: JsValue)
+
+object GetAccountResult {
+  implicit val format: Format[GetAccountResult] = Json.format[GetAccountResult]
+}
+
+case class GetAccountResultEvent(getAccountResult: GetAccountResult, path: String)(implicit hc: HeaderCarrier, appConfig: AppConfig) extends HTSEvent {
+  val value: ExtendedDataEvent = {
+    HTSEvent(appConfig.appName, "GetAccountResult", Json.toJson(getAccountResult), "GetAccountResult", path)
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/models/HTSEvent.scala]: to keep the transaction name consistent with the others, could we have "get-account-result" for the transaction name here?

File: app/uk/gov/hmrc/helptosave/models/HTSEvent.scala [hunk 3]
```diff
@@ -154,3 +154,16 @@ object AccountCreated {
...
 }
+
+case class GetAccountResult(nino: String, account: JsValue)
+
+object GetAccountResult {
+  implicit val format: Format[GetAccountResult] = Json.format[GetAccountResult]
+}
+
+case class GetAccountResultEvent(getAccountResult: GetAccountResult, path: String)(implicit hc: HeaderCarrier, appConfig: AppConfig) extends HTSEvent {
+  val value: ExtendedDataEvent = {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/models/HTSEvent.scala]: there's a `HtsEventSpec` you can add to to add a unit test to check that the fields are written as expected

File: app/uk/gov/hmrc/helptosave/models/HTSEvent.scala (hunk 2 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/models/HTSEvent.scala]: fixed

File: app/uk/gov/hmrc/helptosave/models/HTSEvent.scala (hunk 3 above)
Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/models/HTSEvent.scala]: added unit test

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**